packages/compile-solidity/compileWithPragmaAnalysis.js

```diff
@@ -0,0 +1,139 @@
+const CompilerSupplier = require("./compilerSupplier");
+const Config = require("@truffle/config");
+const semver = require("semver");
+const Profiler = require("./profiler");
+const fse = require("fs-extra");
+const { run } = require("./run");
+const OS = require("os");
+const cloneDeep = require("lodash.clonedeep");
+
+const getSemverExpression = source => {
```

Comment:
Do we want to throw an explicit error message here if one of the files has no version pragma?  I feel like that's worth an explicit error message, saying like, you have to put pragmas in every file if you want to use this feature.  Or is that not a restriction, and files with no pragmas just don't restrict the version used?  (If nothing has any pragmas, do we use latest Solidity?)

Reply:
Are pragma expressions optional in Solidity files? I had no idea you could omit it...

Reply:
Yes, they're optional.  It's a compiler warning, that's all.